frontend/src/Patient/PatientComponents/PatientDashboardPageComponents/TreatmentOverview.jsx

```diff
@@ -1,12 +1,30 @@
-import { treatments, users } from "../../../../data";
+import { useState, useEffect } from "react";
+import { getPatientTreatment } from "../../../api";
+
+function TreatmentOverview({ id }) {
+    const [treatment, setTreatment] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        (async () => {
+            setTreatment(await getPatientTreatment(id));
+            setIsLoading(false);
+        })();
+    }, []);
```

Comment:
[non-blocking] Consider using SWR for API calls, will give you the state + loading + errors + useEffect all in one: https://github.com/vercel/swr

Will make your life much simpler

Reply:
```suggestion
    const fetcher = async () => await getPatientTreatment(id);
    
    const { treatment, error, isLoading } = useSWR('/treatment', fetcher);
```